### ISSUE TITLE
Please allow Bintray links

### ISSUE THREAD

**Issue by probonopd** [opendesktop/issues#4]
(Continued from [here](https://forum.opendesktop.org/t/please-allow-bintray-links/523/1))

In addition to GitHub download links, please also enable Bintray download links.
I have been piling up lots of AppImages on https://bintray.com/probono/AppImages and there is no way I will re-upload all of them. In fact, these get uploaded to Bintray in a fully automated process straight from [Travis CI](https://travis-ci.org/).

**Comment by star-buck** [comment]
Does this "fully automated process" work for github.com "straight from
Travis CI" as well?

**Comment by probonopd** [comment]
Theoretically yes, (for me) practically no:

GitHub Releases is meant for the releases of _one_ project. For each release, one can upload, in addition to the sources that are automatically taken from Git, binary attachments such as deb, rpm, AppImage, you name it.

However, I have one repository that holds recipes for hundreds of different applications for which I provide sample AppImages. All these applications are on the same GitHub repository, because that allows me to very neatly automate the AppImage generation. I could upload all of these to the same GitHub Releases page, but it would be a complete mess in no time, and I am pretty sure I would hit GitHub quotas relatively soon. A GitHub Releases page is not meant to host a whole universe of different applications. Hence I am using a repository at Bintray. In fact, Bintray was kind enough to increase the quota for https://bintray.com/probono/AppImages significantly above what they normally grant to open source repositories.

So in summary, GitHub Releases works great in cases where you have just one application in one project, e.g., [here](https://github.com/probonopd/Emacs.AppImage/). 99% of GitHub repositories hold one application and should be fine.

My use case is an exception.

**Comment by probonopd** [comment]
Theoretically yes, (for me) practically no:

GitHub Releases is meant for the releases of _one_ project. For each release, one can upload, in addition to the sources that are automatically taken from Git, binary attachments such as deb, rpm, AppImage, you name it.

However, I have one repository that holds recipes for hundreds of different applications for which I provide sample AppImages. All these applications are on the same GitHub repository, because that allows me to very neatly automate the AppImage generation. I could upload all of these to the same GitHub Releases page, but it would be a complete mess in no time, and I am pretty sure I would hit GitHub quotas relatively soon. A GitHub Releases page is not meant to host a whole universe of different applications. Hence I am using a repository at Bintray. In fact, Bintray was kind enough to increase the quota for https://bintray.com/probono/AppImages significantly above what they normally grant to open source repositories.

So in summary, GitHub Releases works great in cases where you have just one application in one project, e.g., [here](https://github.com/probonopd/Emacs.AppImage/). 99% of GitHub repositories hold one application and should be fine.

My use case is an exception.

**Comment by star-buck** [comment]
So for a single app developer, how does one set up this full automated process from github to Travis CI?

**Comment by probonopd** [comment]
The usual process is:
* Developer goes to https://travis-ci.org/ and sign in with his GitHub account (once)
* In the upper right corner developer clicks on his name and is presented with the list of his GitHub projects (only the ones with administrative authority). Developer toggles the switch on one of the projects (once)
* Developer creates a `.travis.yml` file in the GitHub repository (once). [Example](https://github.com/probonopd/Emacs.AppImage/blob/master/.travis.yml)
* Developer does a `git push` and pushes a code change to GitHub
* GitHub automatically sends a notification to Travis CI
* Travis CI automatically downloads the code from GitHub
* Travis CI automatically reads the contents of the `.travis.yml` file and executes it. Normally the file contains instructions on how to compile and test the software. In my case, I also run a [script](https://github.com/probonopd/Emacs.AppImage/blob/master/appimage.sh) to make an AppImage (this script could probably be made so general that it covers most of the cases; actually the [linuxdeployqt](https://github.com/probonopd/linuxdeployqt) tool can hopefully replace it soon and do the thing automagically)
* If the build succeeds, Travis CI can automatically upload the build artefacts of tagged releases to GitHub Releases using https://docs.travis-ci.com/user/deployment/releases/
* Since the official GitHub Releases provider above apparently only works on tagged releases, I have written my own script that can do the same for continuous builds, i.e., it creates a new release called "continuous" and uploads the build artefacts there, see https://github.com/probonopd/uploadtool

This is a quick overview. Let me know if something is unclear or you run into trouble.
